pex/common.py

```diff
@@ -363,7 +386,45 @@ def __str__(self):
   def delete(self):
     shutil.rmtree(self.chroot)
 
-  def zip(self, filename, mode='w'):
+  def zip(self, filename, mode='w', deterministic_timestamp=False):
     with open_zip(filename, mode) as zf:
       for f in sorted(self.files()):
-        zf.write(os.path.join(self.chroot, f), arcname=f, compress_type=zipfile.ZIP_DEFLATED)
+        full_path = os.path.join(self.chroot, f)
+        if not deterministic_timestamp:
+          zf.write(
+            full_path,
+            arcname=f,
+            compress_type=zipfile.ZIP_DEFLATED
+          )
+        else:
```

Comment:
This `else` should be broken out as a separate method I think.

Reply:
I thought about doing this and played with it just now, and am not convinced it should be split out. The logic is not generalizable to anything else, which imo is the reason to extract a function. 

We also don't explicitly test every detail of the implementation beyond the overall deterministic timestamp unit test, and I don't think we should because every single line comes directly from CPython's source, modulo the `date_time` line, comments, and parsing the data. So, extracting the function for better testing isn't a motivation imo.

I can split it out if you'd like but think it's best to keep the logic closer to where it's used.

Reply:
Fine with me.